src/hoc/asyncComponent.jsx

```diff
@@ -0,0 +1,24 @@
+import React, { Component } from 'react';
+
+const asyncComponent = (importComponent) => {
+	return class extends Component {
+		state = {
+			component: null
+		}
+
+		componentDidMount() {
+			importComponent()
+				.then(cmp => {
+					this.setState({ component: cmp.default });
+				});
+		}
+
+		render() {
+			const C = this.state.component;
+
+			return C ? <C {...this.props} /> : null;
+		}
+	};
+};
+
+export default asyncComponent;
```

Comment:
This is a pretty cool and interesting HOC. Just curious. What's the purpose behind this? It seems like it's not being used anywhere.

Reply:
maxmillian schwartz's course copy paste
 thought i needed it for routes but found a simpler way i think
screw u maxmillian